In [ ]:
import pandas as pd


In [ ]:
df = pd.read_csv("base_rh.csv", sep=';', encoding='cp1252', decimal=',')
df.info()
df.describe()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ID_Funcionario  1000 non-null   int64  
 1   Nome            1000 non-null   str    
 2   Departamento    1000 non-null   str    
 3   Cargo           1000 non-null   str    
 4   Salario         1000 non-null   float64
 5   Data_Admissao   1000 non-null   str    
 6   Genero          1000 non-null   str    
 7   Idade           1000 non-null   int64  
 8   Estado_Civil    1000 non-null   str    
 9   Status          1000 non-null   str    
dtypes: float64(1), int64(2), str(7)
memory usage: 78.3 KB


,ID_Funcionario,Salario,Idade
count,1000.000000,1000.000000,1000.000000
mean,500.500000,8579.951780,41.399000
std,288.819436,3657.365307,13.665817
min,1.000000,2000.710000,18.000000
25%,250.750000,5564.545000,30.000000
50%,500.500000,8571.135000,41.000000
75%,750.250000,11554.630000,53.000000
max,1000.000000,14954.510000,65.000000


### Informações do DataFrame (df.info())

```text
RangeIndex: 1000 entries, 0 to 999
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ID_Funcionario  1000 non-null   int64  
 1   Nome            1000 non-null   str    
 2   Departamento    1000 non-null   str    
 3   Cargo           1000 non-null   str    
 4   Salario         1000 non-null   float64
 5   Data_Admissao   1000 non-null   str    
 6   Genero          1000 non-null   str    
 7   Idade           1000 non-null   int64  
 8   Estado_Civil    1000 non-null   str    
 9   Status          1000 non-null   str    
dtypes: float64(1), int64(2), str(7)
```

| Estatística | ID_Funcionario | Salario | Idade |
|:---|:---:|:---:|:---:|
| **count** | 1000.00 | 1000.00 | 1000.00 |
| **mean** | 500.50 | 8579.95 | 41.40 |
| **std** | 288.82 | 3657.37 | 13.67 |
| **min** | 1.00 | 2000.71 | 18.00 |
| **25%** | 250.75 | 5564.55 | 30.00 |
| **50%** | 500.50 | 8571.14 | 41.00 |
| **75%** | 750.25 | 11554.63 | 53.00 |
| **max** | 1000.00 | 14954.51 | 65.00 |


In [ ]:
"""3. Converta `Data_Admissao` para datetime. Crie as colunas:  
    `Ano_Admissao`, `Mes_Admissao`, `Tempo_Casa_Anos` e `Faixa_Tempo_Casa`.
"""

df['Data_Admissao'] = pd.to_datetime(df['Data_Admissao'], format='%d/%m/%Y', errors='coerce')
df['Ano_Admissao'] = df['Data_Admissao'].dt.year
df['Mes_Admissao'] = df['Data_Admissao'].dt.month
df['Tempo_Casa_Anos'] = ((pd.Timestamp.today() - df['Data_Admissao']).dt.days / 365.25).round(2)
df['Faixa_Tempo_Casa'] = pd.cut(
    df['Tempo_Casa_Anos'],
    bins=[-1, 1, 3, 5, 10, 100],
    labels=['0-1', '1-3', '3-5', '5-10', '10+']
)

df.head()

In [ ]:
"""4. Responda com groupby,
    "→ Qual departamento tem a **maior** média de tempo de casa?,
    "→ Qual tem a **menor** média de tempo de casa?,
"""

tempo_medio = df.groupby('Departamento')['Tempo_Casa_Anos'].mean().sort_values()
print('Departamento com menor tempo médio de casa:', tempo_medio.idxmin())
print('Departamento com maior tempo médio de casa:', tempo_medio.idxmax())

tempo_medio

In [ ]:
from pathlib import Path

output_file = Path(r"C:\Users\jakso_sdyff07\OneDrive\Documentos\git_sc_tec\turma-visualizacao-de-dados\alunos\jakson_luis\exercicios\semana_4\base_rh_deptos.xlsx")
output_file.parent.mkdir(parents=True, exist_ok=True)

with pd.ExcelWriter(output_file) as writer:
    df.to_excel(writer, sheet_name='Completo', index=False)
    for dept in df['Departamento'].unique():
        df[df['Departamento'] == dept].to_excel(writer, sheet_name=dept, index=False)

print(f"Arquivo salvo em: {output_file}")